doc/source/cluster/kubernetes/examples/rayjob-kueue-gang-scheduling.md

```diff
@@ -0,0 +1,249 @@
+* Kueue admits the RayJob, and its execution begins with Ray nodes scheduled on the newly provisioned nodes.
+
+In the event that GPUs are unavailable, Kueue will keep the RayJob suspended. In addition, the node autoscaler will avoid provisioning any new nodes unless it can fully satisfy the RayJob's GPU requirements.
+
+Upon creating a RayJob, notice that the RayJob is immediately suspended despite the ClusterQueue having GPU quotas available.
+```bash
+$ kubectl get rayjob pytorch-text-classifier-rj4sg -o yaml
+apiVersion: ray.io/v1
+kind: RayJob
+metadata:
+  name: pytorch-text-classifier-rj4sg
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+...
+...
+...
+status:
+  jobDeploymentStatus: Suspended  # RayJob suspended
+  jobId: pytorch-text-classifier-rj4sg-pj9hx
+  jobStatus: PENDING
+```
+
+Kueue will keep this RayJob suspended until it's corresponding ProvisioningRequest is satisfied.
+You can list ProvisioningRequest resources and their status:
+```bash
+$ kubectl get provisioningrequest
+NAME                                                      ACCEPTED   PROVISIONED   FAILED   AGE
+rayjob-pytorch-text-classifier-nv77q-e95ec-rayjob-gpu-1   True       False         False    22s
+```
+
+Note there are two coloumns in the output: `ACCEPTED` and `PROVISIONED`.
+`ACCEPTED=True` means that Kueue and the node autoscaler has acknowledged the request.
+`PROVISIONED=True` means that the node autoscaler has completed provisioning nodes.
+Once both of these conditions are true, the ProvisioningRequest is considered satisfied.
+```bash
+$ kubectl get provisioningrequest
+NAME                                                      ACCEPTED   PROVISIONED   FAILED   AGE
+rayjob-pytorch-text-classifier-nv77q-e95ec-rayjob-gpu-1   True       True          False    57s
+```
+
+Because the example RayJob requires 1 GPU for fine-tuning, the ProvisioningRequest is satisfied
+by the addition of a single GPU node in the `gpu-node-pool` Node Pool.
+```bash
+$ kubectl get nodes
+NAME                                                  STATUS   ROLES    AGE   VERSION
+gke-kuberay-gpu-cluster-default-pool-8d883840-fd6d    Ready    <none>   14m   v1.29.0-gke.1381000
+gke-kuberay-gpu-cluster-gpu-node-pool-b176212e-g3db   Ready    <none>   46s   v1.29.0-gke.1381000
+```
+
+
+Once the ProvisioningRequest is satisfied, Kueue admits the RayJob.
+The Kubernetes scheduler then immediately places the head and worker nodes onto the newly provisioned resources.
+The ProvisioningRequest ensures a seamless Ray cluster start up, with no scheduling delays for any pods.
+
+```bash
+$ kubectl get pods
+NAME                                                      READY   STATUS    RESTARTS        AGE
+pytorch-text-classifier-nv77q-g6z57                       1/1     Running   0               13s
+torch-text-classifier-nv77q-raycluster-gstrk-head-phnfl   1/1     Running   0               6m43s
+```
+
+## Why use gang scheduling?
```

Comment:
Perhaps we should move it to the beginning of the document. "Why" is more important than "How".

Reply:
Good point, I'll move it